historical/common/dynamodb.py

```diff
@@ -99,54 +93,73 @@ def deserialize_current_dynamo_to_pynamo(record, model):
     :param model:
     :return:
     """
+    # Was the item in question too big for SNS? If so, then we need to fetch the item from the current Dynamo table:
+    if record.get("lol_too_big"):
+        arn = record['dynamodb']['Keys']['arn']['S']
+        log.debug('[-->] Item with ARN: {} was too big for SNS -- fetching it from the Current table...'.format(arn))
+        item = list(current_model.query(arn))
+
+        # If for whatever reason, the item *cannot* be found, then this record should be skipped over.
+        # This will happen if this event came in and got processed right after the item was deleted
+        # from the Current table. If so, then do nothing -- the deletion event will be processed later.
+        if not item:
+            return None
+
+        # We need to place the real configuration data into the record so it can be deserialized into
+        # the durable model correctly:
+        record['dynamodb']['NewImage']['configuration'] = item[0]._serialize()['attributes']['configuration']
+
     new_image = remove_current_specific_fields(record['dynamodb']['NewImage'])
     data = {}
 
     for item in new_image:
         # This could end up as loss of precision
         data[item] = deser.deserialize(new_image[item])
 
-    return model(**data)
+    return durable_model(**data)
 
 
-def process_dynamodb_differ_record(record, durable_model, diff_func=None, region_attr="Region"):
+def process_dynamodb_differ_record(record, current_model, durable_model, diff_func=None):
     """
-    Processes a group of DynamoDB NewImage records (for Differ events).
+    Processes a DynamoDB NewImage record (for Differ events).
 
-    This will ONLY process the record if the record exists in one of the regions defined in the
-    DIFF_REGIONS environment variable.
+    This will ONLY process the record if the record exists in one of the regions defined by the SNSZR_REGIONS of the
```

Comment:
Its not clear via the variable name what the `ZR` part or `SNSZR_REGIONS` is.

Reply:
Renaming it to `SQSProxy`.